state-chain/chains/src/btc/smart_contract_encoding.rs

```diff
@@ -0,0 +1,95 @@
+use crate::ForeignChainAddress;
+use cf_primitives::{Asset, AssetAmount};
+use codec::{Decode, Encode};
+
+use super::{BitcoinOp, BitcoinScript};
+
+// The maximum length of data that can be encoded in a nulldata utxo
+const MAX_NULLDATA_UTXO_LENGTH: usize = 80;
+
+#[derive(Encode, Decode, Clone)]
+pub struct UtxoEncodedData {
+	pub output_asset: Asset,
+	pub output_address: ForeignChainAddress,
+	pub parameters: SharedCfParameters,
+}
+
+// The encoding of these parameters is the same across chains
+#[derive(Encode, Decode, Clone)]
```

Comment:
```suggestion
#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone)]
```

We will need these too.